### PR TITLE
Rename define-simple-macro from syntax/parse/define to define-syntax-parse-rule

### PR DESCRIPTION
**Pull request by lexi-lambda**
In [a mailing list thread from last year](https://groups.google.com/d/msg/racket-dev/e4WfuZ9JWUk/StzZW7lBCQAJ), it was mentioned that `define-simple-macro` might not be the best name for the `define-syntax-rule` analog exported from `syntax/parse/define`. I was not super satisfied with any of the proposed alternate names, but after considering it for a while, I think `define-syntax-pattern` is a fair one.
1. It is pretty close to `define-syntax-rule`, but distinct enough to not be any more confusing than `syntax-case` vs `syntax-parse`.
2. The term `pattern` is already used a lot within the `syntax/parse` docs, so it is a familiar term (probably more than `rule`, really).
3. Even in isolation, the name pretty accurately describes what the form does.

Obviously, I’ve left the old name in to avoid breaking backwards compatibility, but I’ve updated the docs to mention that the other name is preferred.

### CODE REVIEW

**Comment by stamourv** [comment]
I'm not sold on the name. `pattern` describes only half of what `define-simple-macro` does (`template` being the other half).
Some more ideas: `define-syntax-rewrite`, `define-syntax-clause`.

**Comment by jackfirth** [comment]
Would `define-syntax-parse-rule` make sense?

**Comment by mbutterick** [comment]
> I'm not sold on the name.

Nor I. A syntax pattern is a distinct thing in Racket (and it's already nettlesome that `syntax/parse`  supports syntax patterns that `syntax-case` does not, so to introduce a third meaning of the phrase ...)

> `define-syntax-parse-rule`

= best, by analogy to `define-syntax-rule`

**Comment by lexi-lambda** [comment]
I’d be fine with `define-syntax-parse-rule`. It’s a bit wordy, but it’s probably the only name I’ve seen so far without any significant problems. Any opposed?

**Comment by stamourv** [comment]
While we're at it, how about we make this new form implicitly quasisyntax its template? (as opposed to syntax, which is what `define-simple-macro` does).
I just wrote a bunch of code that used `define-simple-macro` and explicitly quasisyntaxed the template, which it turns out does not work. Now I have to use the full-fledged `syntax-parse` instead, which is a bummer. Quasisyntaxing would have solved my problem.

**Comment by lexi-lambda** [comment]
Why not take it a step further? What about using `template` from `syntax/parse/experimental/template` instead of `syntax` or `quasisyntax`? And while we’re at it, what about getting that out of its `experimental` directory?

**Comment by jeapostrophe** [comment]
+1

On Wednesday, July 6, 2016, Alexis King <notifications@github.com> wrote:

> Why not take it a step further? What about using template from
> syntax/parse/experimental/template instead of syntax or quasisyntax? And
> while we’re at it, what about getting that out of its experimental
> directory?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/racket/racket/pull/1310#issuecomment-230921279>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAOM-u2rSemxCSXHYgXpFejW-4IKNA9Xks5qTCZ1gaJpZM4IIz39>
> .
>


-- 
Jay McCarthy
Associate Professor
PLT @ CS @ UMass Lowell
http://jeapostrophe.github.io

           "Wherefore, be not weary in well-doing,
      for ye are laying the foundation of a great work.
And out of small things proceedeth that which is great."
                          - D&C 64:33

**Comment by stamourv** [comment]
Sounds great!
@rmculpepper: Any blockers to moving templates out of experimental?

**Comment by sorawee** [comment]
Should we revisit this PR? The "moving templates out of experimental" is already done since the initial release of Racket 7 IIUC. 

There's a conflict that must be resolved though.

**Comment by jackfirth** [comment]
I think we should revisit this, yeah. Every time I write `define-simple-macro` my eye twitches a little.

@sorawee It might be easier to just make a new PR than to try and resolve this one's conflicts. It's not exactly a complex change.
